### PR TITLE
Replace manual rsync by docker-rsync

### PR DESCRIPTION
**Pull request by ajardin**
Even if the manual rsync brings some (really) nice performance improvements, it can be quite annoying to use on a day-to-day basis... So, I propose to replace it by [docker-rsync](https://docker-sync.readthedocs.io/en/latest/) which is fully transparent once started.

**Installation**
* `gem install docker-sync`
* `brew install unison eugenmayer/dockersync/unox rsync`

**Usage** 
* `make start` and `make stop` as usual

### CODE REVIEW

File: docker-sync.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  cli_mode: 'auto'
+  project_root: 'pwd'
+
+syncs:
+  app-magento2-sync:
+    src: ${PROJECT_LOCATION}
+    host_disk_mount_mode: 'cached'
+
+    sync_userid: 1000
+    sync_excludes: ['.git']
```
Comment (maxhelias) [inline docker-sync.yml]: I think you can also add the `.idea` folder

File: docker-sync.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    sync_excludes_type: 'Name'
+
```
Comment (maxhelias) [inline docker-sync.yml]: You can also add this line : `sync_strategy: ${DOCKER_SYNC_STRATEGY}` if someone want to change the strategy

**Comment by maxhelias** [comment]
I would add two non-blocking point : 
 - I don't know how works `notify_terminal: true` but that would be great
 - And on our volumes nginx & php, i would use `${PROJECT_LOCATION}` and add the option `compose-dev-file-path` on docker-sync.yml to load a specific file only on dev mode for creating the volume `app-magento2-sync`

It is this operation that is used for this project (i have used in the past) : https://github.com/laradock/laradock

WDYT ?

File: docker-sync.yml (hunk 1 above)
Comment (ajardin) [reply docker-sync.yml]: I wanted to avoid adding such specific things, but why not. :wink:

File: docker-sync.yml (hunk 2 above)
Comment (ajardin) [reply docker-sync.yml]: Changing the configuration will imply to configure other parameters. In consequence, I don't think it's a good idea to allow this change through a unique variable. Let's keep it as it for now.

**Comment by ajardin** [comment]
Thanks for the suggestions!

The terminal notification doesn't work on my computer. Maybe you'll have better chances with it: the parameter is now enabled. And there is no need to use the `compose-dev-file-path` since we are discussing about something dedicated to local environments only.

**Pull request merged**